src/sagemaker/serverless/predictor.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2017-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Predictors that are hosted on serverless compute."""
+from __future__ import absolute_import
+
+from typing import Optional, Tuple
+
+import boto3
+import botocore
+
+from sagemaker import deserializers, serializers
+from sagemaker.predictor import PredictorBase
+
+
+class LambdaPredictor(PredictorBase):
+    """A deployed model hosted on Lambda."""
+
+    def __init__(
+        self, function_name: str, client: Optional[botocore.client.BaseClient] = None
+    ) -> None:
+        """Initialize instance attributes.
+
+        Arguments:
+            function_name: The name of the function.
+            client: The Lambda client used to interact with Lambda.
+        """
+        self._client = client or boto3.client("lambda")
+        self._function_name = function_name
+        self._serializer = serializers.JSONSerializer()
+        self._deserializer = deserializers.JSONDeserializer()
+
+    def predict(self, data: dict) -> dict:
+        """Invoke the Lambda function specified in the constructor.
+
+        This function is synchronous. It will only return after the function
+        has produced a prediction.
+
+        Arguments:
+            data: The data sent to the Lambda function as input.
+
+        Returns:
+            The data returned by the Lambda function.
+        """
+        response = self._client.invoke(
```

Comment:
Did we decide not to support any sort of error handling even for retryable exceptions?

Reply:
Yeah, that's correct. If there's an error, we'll just pass up whatever Boto throws